src/bin/pg_autoctl/systemd_config.c

```diff
@@ -108,8 +124,8 @@ systemd_config_init(SystemdServiceConfig *config, const char *pgdata)
 	sformat(config->EnvironmentPGDATA, BUFSIZE,
 			"'PGDATA=%s'", config->pgSetup.pgdata);
 
-	/* adjust the user to the current system user */
-	strlcpy(config->User, user, NAMEDATALEN);
+	/* adjust the user to the owner of PGDATA */
+	strlcpy(config->User, pw->pw_name, NAMEDATALEN);
```

Comment:
Maybe also set `config->Group`?

Reply:
Mmm, there's no “Group” in `SystemdServiceConfig` right?

Reply:
I mean it's our own type right? So we can add it. I meant it more like, maybe we should set both `User` and `Group` in the final systemd config.

Reply:
Oh yeah, I didn't know you could set the group for systemd and then missed your comment entirely, sorry about that. About the group, given Unix design where a user can belong to many groups and a directory only to one single group, and given how Postgres is packaged for debian and some other OS, I'm not sure it would be a good idea for us to do that, or at least I have to dive into more details.

A common security hardening guideline consists of using your own Postgres system super-user that's not named “postgres” on the system, and belongs to the “postgres” group. What would be the Systemd Group to chose there? What would be the PGDATA group set to when we call initdb in such a setup? I just don't know yet... if you think we should spend time on that aspect, I will and then open a new PR with my findings...